src/fw_update.c

```diff
@@ -52,6 +52,11 @@ static enum golioth_status fw_write_block_cb(const struct golioth_ota_component
                                                         negotiated_block_size * block_idx,
                                                         component->size);
 
+    if (status != GOLIOTH_OK)
+    {
+        return status;
+    }
+
     ctx->bytes_downloaded += block_buffer_len;
```

Comment:
We can make this clearer by making sure there's only one return statement in this function. Either changing this `return` into ` goto`, or removing the added `if` and instead wrapping the rest  of the function in `if (status == GOLIOTH_OK)`:

```
if (status == GOLIOTH_OK)
{
    ctx->bytes_downloaded += block_buffer_len;

    if (is_last)
    {
        fw_update_post_download();
    }
}

return status;
```